satpy/tests/reader_tests/modis_tests/_modis_fixtures.py

```diff
@@ -243,17 +241,27 @@ def create_hdfeos_test_file(filename: str,
         file_shortname: Short name of the file to be stored in global metadata
             attributes. Only used if ``include_metadata`` is ``True``
             (default).
-        include_metadata: Include global metadata attributes (default: True).
+        metadata_dict: A dictionary of metadata to be added to the file.
 
     """
     h = SD(filename, SDC.WRITE | SDC.CREATE)
 
-    if include_metadata:
-        if geo_resolution is None or file_shortname is None:
-            raise ValueError("'geo_resolution' and 'file_shortname' are required when including metadata.")
-        setattr(h, 'CoreMetadata.0', _create_core_metadata(file_shortname))  # noqa
-        setattr(h, 'StructMetadata.0', _create_struct_metadata(geo_resolution))  # noqa
-        setattr(h, 'ArchiveMetadata.0', _create_header_metadata())  # noqa
+    if metadata_dict is not None and metadata_dict != {}:
```

Comment:
I'm not sure this has to be optional at all. I wasn't thinking it'd be the file_shortname and stuff, it'd be instead:


```
metadata = {
    "StructMetadata.0": _create_struct_metadata(geo_res),
    "CoreMetadata.0": _create_core_metadata(file_shortname),
    "ArchiveMetadata.0": _create_header_metadata(),
}
create_hdfeos_test_file(filename, variable_infos, metadata)
```

And `metadata` would be a positional argument (not optional) and could be passed as `{}` for any files that need no metadata.

A whole other idea would be to make the struct/geo metadata, core metadata, and archive metadata individual keyword arguments. That way the user/caller doesn't need to know the exact "StructMetadata.0" attribute name which they could easily screw up.